### PR TITLE
#1863 Simplify field template and unify config layout

### PR DESCRIPTION
**Pull request by BLoe**
Part of #1863. This is an iterative improvement on some layout spacing and styling in the page editor node config panel. There will be a followup PR that further refactors `ObjectWidget` and maybe `ArrayWidget` to address issues with nested fields.

image

Output label never overlaps:
image

### CODE REVIEW

**Comment by fregante** [comment]
> Output label never overlaps:

:chefskiss:

File: src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss [hunk 1]
```diff
@@ -24,6 +24,11 @@
   width: auto;
   flex-grow: 0;
   flex-shrink: 0;
+
+  > button {
+    padding-left: 0.375rem !important;
+    padding-right: 0.375rem !important;
```
Comment (fregante) [inline src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss:30]: Why so specific?

Comment (BLoe) [reply src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss:30]: The bootstrap button defaults to `0.75rem` on the left and right, and `0.375rem` on top and bottom. So, I just made the horizontal padding match the vertical padding 🤷‍♂️

**Review by BALEHOK — APPROVED** [review]

File: src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss (hunk 1 above)
Comment (twschiller) [reply src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss:30]: This is a good candidate for a code comment

**Comment by BLoe** [comment]
This will impact the web app, need to check into that and probably rethink some things here.

**Comment by twschiller** [comment]
> This will impact the web app, need to check into that and probably rethink some things here.

@BLoe what are you worried about in the application?

**Comment by BLoe** [comment]
> @BLoe what are you worried about in the application?

I noticed the PR from Umair that you merged yesterday had some stuff using the `layout` prop in `FieldTemplate`.

File: src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss (hunk 1 above)
Comment (fregante) [reply src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss:30]: Possible nits:

- use a variable
- use `padding: XXX` to change all at once so the intent is well defined
- also drop the `width: 38px` on the `span` inside the `button` to make it a bit smaller (I'm not sure if this causes misalignments elsewhere)

General note about sizing/spacing:

- flexbox is amazing and `stretch` lets us match the heights more safely. For example in this case you could set `padding: 0.3em` and still have the field and button have the same height.
- I feel that we should drop bootstrap’s **columns** (only) in most situations and use flexboxes to size content

Related:

- https://github.com/pixiebrix/pixiebrix-extension/issues/2085
- https://github.com/pixiebrix/pixiebrix-extension/issues/1234
- https://github.com/pixiebrix/pixiebrix-extension/issues/985

File: src/options/pages/services/ServiceEditorModal.module.scss [hunk 2]
```diff
@@ -24,3 +24,9 @@
   // Modal is 980
   z-index: 980 !important;
 }
+
+.dialog {
+  width: 55vw;
+  max-width: 700px;
+  min-width: 400px;
```
Comment (fregante) [inline src/options/pages/services/ServiceEditorModal.module.scss:31]: This is probably what we should aim for:

```suggestion
.dialog {
  width: 100%; // maybe not needed at all
  max-width: 700px;
```

I think it never makes sense to target a _percentage of the width_ when showing a single item on screen.

File: src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss (hunk 1 above)
Comment (BLoe) [reply src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss:30]: I think the main issue I have with dropping use of bootstrap columns is that it's how the `FieldTemplate` component lines up the labels and inputs for basically the entire page editor. Is there a way to do this kind of "column" alignment using just flexbox? If we dropped bootstrap columns would we have to rebuild `FieldTemplate` using flex-grid or something?

Comment (BLoe) [reply src/components/fields/schemaFields/widgets/TemplateToggleWidget.module.scss]: And, good point about the span width, that was left over from when the symbols were just text and not icons, some of them were wider. I reduced the size of it a bit.

**Pull request merged**